src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs

```diff
@@ -7652,5 +7652,367 @@ public void RecordDeclaration()
+                "T P",
+                SymbolDisplayPartKind.TypeParameterName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+        }
+
+        [Fact]
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        public void NullableAttributes_MayBeNullOnStructProperty()
+        {
+            var source = @"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+public class A<T>
+{
+    [MaybeNull]
+    public T P { get; }
+}
+
+public class Test: A<int> { }
+";
+            var compilation = CreateCompilation(new[] { source, MaybeNullAttributeDefinition });
+            var closedGenericMember = compilation.GetTypeByMetadataName("Test").BaseType().GetMember("P");
+            var format = new SymbolDisplayFormat(
+                memberOptions: SymbolDisplayMemberOptions.IncludeParameters | SymbolDisplayMemberOptions.IncludeType | SymbolDisplayMemberOptions.IncludeModifiers,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier);
+
+            Verify(
+                closedGenericMember.ToDisplayParts(format),
+                "Int32 P",
+                SymbolDisplayPartKind.StructName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName);
+        }
+
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        [Fact]
+        public void NullableAttributes_VisualBasicReferencesCSharp()
+        {
+            var sourceCS = @"
+using System.Diagnostics.CodeAnalysis;
+
+#nullable enable
+
+namespace N
+{
+    public class A<T>
+    {
+        [MaybeNull]
+        public T P { get; }
+    }
+}
+";
+            var sourceVB = @"
+Imports N
+Public Class Test
+    Inherits A(Of Object)
+End Class";
+            var compCS = CreateCompilation(new[] { sourceCS, MaybeNullAttributeDefinition });
+            var refCS = compCS.EmitToImageReference();
+            var compVB = CreateVisualBasicCompilation(GetUniqueName(), sourceVB, referencedAssemblies: new[] { MscorlibRef, refCS });
+            var closedGenericMember = compVB.GetTypeByMetadataName("Test").BaseType.GetMember("P");
+            var openGenericMember = compVB.GetTypeByMetadataName("N.A`1").GetMember("P");
+            var format = new SymbolDisplayFormat(
+                memberOptions: SymbolDisplayMemberOptions.IncludeParameters | SymbolDisplayMemberOptions.IncludeType | SymbolDisplayMemberOptions.IncludeModifiers,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier);
+
+            Verify(
+                closedGenericMember.ToDisplayParts(format),
+                "Overloads P As Object",
```

Comment:
>"Overloads P As Object", [](start = 16, length = 24)

This isn't quite what I had in mind. It looks like we are testing VB's SymbolDisplay, which this PR doesn't change. Even though it is good to have a test like this for completeness, it is more interesting to test behavior of C#'s SymbolDisplay when it is fed VB symbols.

Reply:
I added `NullableAttributes_VisualBasicReferencesCSharp` (this test) and `NullableAttributes_CSharpReferencesVisualBasic`, so there is a test for the other way around.

Reply:
> I added NullableAttributes_VisualBasicReferencesCSharp (this test) and NullableAttributes_CSharpReferencesVisualBasic, so there is a test for the other way around.

The ```NullableAttributes_CSharpReferencesVisualBasic``` tests something else. It looks like it tests C# SymbolDisplay for C# symbols representing members originally declared in VB. In my opinion, this isn't particularly interesting since VB declarations in the test don't do anything that cannot be done with declarations in C#. The test that we are missing is C# SymbolDisplay for VB symbol in this test. See how ```TestMethodVB``` in this file achieves that. We want to test changes that you are making in SymbolDisplayVisitor.Members.cs with VB symbols as an input.

---
In reply to: [561243985](https://github.com/dotnet/roslyn/pull/50527#discussion_r561243985) [](ancestors = 561243985)

Reply:
Sorry for getting this wrong and thanks for the pointer to a sample test. I added a new test in 569d0ea7 that should test the right thing now. One kind of symbol, that is not covered yet are `SignatureOnly...` symbols. Azure pipelines CI revealed that (see [test-results](https://dev.azure.com/dnceng/public/_build/results?buildId=957883&view=ms.vss-test-web.build-test-results-tab&runId=30203184&resultId=107662&paneView=debug)). I fixed that in 388097d. Do we need tests for this kind of symbol too?

Reply:
> I added a new test in 569d0ea7 that should test the right thing now.

NullableAttributes_SymbolDisplayForVBSymbol is good to have. However, I would like us to have VB symbols from **this test** pushed through C# SymbolDisplay. It is interesting to see how the same members declared in C# displayed differently or not differently when looked at through VB symbol model. Sorry for not making this clear from the beginning.

---
In reply to: [561753871](https://github.com/dotnet/roslyn/pull/50527#discussion_r561753871) [](ancestors = 561753871)

Reply:
> One kind of symbol, that is not covered yet are ```SignatureOnly...``` symbols. 

I am somewhat surprised that ```SignatureOnly...``` symbols are leaked through diagnostics. Perhaps this (the leak) is something we should take a closer look at. Could you find out what specific error does that in the identified test? Since now we a "losing" control over usage of ```FlowAnalysisAnnotations``` API, I think we need to make sure that it doesn't throw without a good reason. Please review all implementations of this API and make sure we no longer assume that the code is unreachable. Then, I would be comfortable with not having targeted SymbolDisplay tests for all different implementations of the API. 

---
In reply to: [561978351](https://github.com/dotnet/roslyn/pull/50527#discussion_r561978351) [](ancestors = 561978351,561753871)

Reply:
I see. I think f122b80 is what you were asking for.

Reply:
> I am somewhat surprised that SignatureOnly... symbols are leaked through diagnostics. Could you find out what specific error does that in the identified test?

`SignatureOnlyParameterSymbol` is used to improve the diagnostic message in overload resolution (note: the list all classes wich construct SignatureOnlyParameterSymbol, but not all places, I have not looked for other `SignatureOnly..` symbols):

https://github.com/dotnet/roslyn/blob/ff498f2fd470b6ff2b956ac0c4317667281688ad/src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs#L1260-L1282

There are also other places where `SignatureOnlyParameterSymbol` is created, that do not seem to be related to diagnostics:

Binder_Crefs
https://github.com/dotnet/roslyn/blob/ff498f2fd470b6ff2b956ac0c4317667281688ad/src/Compilers/CSharp/Portable/Binder/Binder_Crefs.cs#L885-L900

`dynamic` handling in Conversions.cs
https://github.com/dotnet/roslyn/blob/ff498f2fd470b6ff2b956ac0c4317667281688ad/src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs#L250-L272

Retargeting:
https://github.com/dotnet/roslyn/blob/ff498f2fd470b6ff2b956ac0c4317667281688ad/src/Compilers/CSharp/Portable/Symbols/Retargeting/RetargetingSymbolTranslator.cs#L1030-L1042

Records member synthesizing:
https://github.com/dotnet/roslyn/blob/ff498f2fd470b6ff2b956ac0c4317667281688ad/src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs#L3140-L3157

Reply:
> Please review all implementations of this API and make sure we no longer assume that the code is unreachable. Then, I would be comfortable with not having targeted SymbolDisplay tests for all different implementations of the API.

There are 8 types, that override the abstract `ParameterSymbol.FlowAnalysisAnnotations` property. `SymbolOnlyParameterSymbol` was the only one, that was throwing. All others either return `None` (FunctionPointerParameterSymbol, SignatureOnlyParameterSymbol, SourceClonedParameterSymbol, SourceSimpleParameterSymbol, SynthesizedParameterSymbolBase, ThisParameterSymbol), delegate to the underlying symbol (WrappedParameterSymbol) or decode the information (SourceComplexParameterSymbol).

The property itself is exclusively used directly in the NullableWalker (and in `SourceMemberContainerTypeSymbol.CheckValidNullableMethodOverride`). All other accesses are through `SymbolExtensions.GetFlowAnalysisAnnotations`.  I would not expect `SymbolExtensions.GetFlowAnalysisAnnotations` to throw in any case.

I only analyzed ParameterSymbol. If you like, I good do the same for the other affected symbols.